PennMobile/General/SwiftUI Views/CustomPopupView.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  CustomPopupView.swift
+//  PennMobile
+//
+//  Created by Christina Qiu on 1/26/24.
+//  Copyright © 2024 PennLabs. All rights reserved.
+//
+
+#if canImport(SwiftUI)
+import SwiftUI
+import PennMobileShared
+#endif
+
+// Global state for the popup
+class PopupManager: ObservableObject {
+    @Published var isShown: Bool = false
+    @Published var image: Image = Image(systemName: "star")
+    @Published var title: String = ""
+    @Published var message: String = ""
+    // Add more properties as needed for buttons and actions
+}
+// Popup View
+struct CustomPopupView: View {
+    @Binding var isShown: Bool
+    var image: Image
+    var title: String
+    var message: String
+    var confirmAction: () -> Void
+    var body: some View {
+        ZStack {
+            // Scrim
+            Color(hex: 0x5F5F64)
+                .opacity(0.2)
+                .edgesIgnoringSafeArea(.all)
+            
+            VStack(spacing: 18) {
+                // Optional image
+                image
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 50, height: 50)
+                    .foregroundColor(.blue)
+                Text(title).font(.headline)
+                Text(message)
+                    .font(.subheadline)
+                    .multilineTextAlignment(.leading)
+                Button(action: confirmAction) {
+                    Text("Confirm")
+                        .foregroundColor(.white)
+                        .padding()
+                        .frame(maxWidth: 300)
+                        .background(RoundedRectangle(cornerRadius: 50).fill(Color.blue))
```

Comment:
Use `Color("navigation")` or define a new custom color in the asset catalog instead? This doesn't quite match the blue in the Figma

![Screenshot 2024-02-03 at 17 10 36](https://github.com/pennlabs/penn-mobile-ios/assets/7005789/ea48fb0c-6dbc-4de1-a1d1-1d81d657abee)

Reply:
Same for the icon above